### PR TITLE
Added a security context to run as non-root user.

### PR DESCRIPTION
**Pull request by NerdSec**
Hello Everyone,

As discussed on the forum with @manics, raising a PR for fixing the security context of the `user-placeholder` pods.

This is my first of many (hopefully) PR on an OSS project. Do let me know in case I missed something.

Regards,
Nachiket

### CODE REVIEW

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/restricted-psp-z2jh-fails-in-local-gitlab-oauth-workflow/6202/5

**Comment by consideRatio** [comment]
Welcome and thanks for being an active community member @NerdSec ! :heart: :tada:

Could you help me review #1798 perhaps? I recently opened a PR that this duplicates in order to resolve this kind of issues for all the Helm chart's pods!

File: jupyterhub/templates/scheduling/user-placeholder/statefulset.yaml [hunk 1]
```diff
@@ -47,4 +47,6 @@ spec:
           image: {{ .Values.prePuller.pause.image.name }}:{{ .Values.prePuller.pause.image.tag }}
           resources:
             {{- include "jupyterhub.resources" . | nindent 12 }}
+      securityContext:
+        runAsUser: 1000
```
Comment (consideRatio) [inline jupyterhub/templates/scheduling/user-placeholder/statefulset.yaml:51]: This is a podSecurityContext, which can influence sidecar containers for example from `istio` which I've seen can lead to issues before, so I think we should set it on the container level only influencing the containers we actually manage.

Comment (NerdSec) [reply jupyterhub/templates/scheduling/user-placeholder/statefulset.yaml:51]: Yes, this does make sense. Adding it as a `podSecurityContext` seems better than for the entire deployment. Maybe we can add the `runAsGroup` parameter as well to ensure that any changes to the image does not affect us in the future?

The indentation in the code above seems to miss a tab, I think it should be as follows:

```
  image: {{ .Values.prePuller.pause.image.name }}:{{ .Values.prePuller.pause.image.tag }}
  resources:
    {{- include "jupyterhub.resources" . | nindent 12 }}
  securityContext:
    runAsGroup: 1000
    runAsUser: 1000
```

Comment (consideRatio) [reply jupyterhub/templates/scheduling/user-placeholder/statefulset.yaml:51]: Thanks for thinking about this @NerdSec ! I've made it a container securityContext in #1798, and also added runAsGroup there, what do you think?

**Pull request closed without merging**